pkg/gadgets/trace/network/tracer/tracer.go

```diff
@@ -256,27 +247,27 @@ func (t *Tracer) Pop() ([]*types.Event, error) {
 	return events, nil
 }
 
-func (t *Tracer) releaseLink(key string, l *link) {
-	unix.Close(l.sockFd)
-	l.networkGraphObjects.Close()
-	delete(t.attachments, key)
+func (t *Tracer) releaseAttachment(netns uint64, a *attachment) {
+	unix.Close(a.sockFd)
+	a.networkGraphObjects.Close()
+	delete(t.attachments, netns)
 }
 
-func (t *Tracer) Detach(key string) error {
-	if l, ok := t.attachments[key]; ok {
-		l.users--
-		if l.users == 0 {
-			t.releaseLink(key, l)
+func (t *Tracer) Detach(pid uint32) error {
+	for netns, a := range t.attachments {
+		if _, ok := a.users[pid]; ok {
+			delete(a.users, pid)
+			if len(a.users) == 0 {
+				t.releaseAttachment(netns, a)
+			}
+			return nil
 		}
-		return nil
-	} else {
-		return fmt.Errorf("key not attached: %q", key)
 	}
+	return fmt.Errorf("pid %d is not attached", pid)
 }
 
 func (t *Tracer) Close() {
 	for key, l := range t.attachments {
-		t.releaseLink(key, l)
+		t.releaseAttachment(key, l)
 	}
-	t.networkGraphMapObjects.Close()
```

Comment:
Why to remove this line? This is created in `NewTracer` and won't be released by `releaseAttachment`.

Reply:
Oops, it wasn't on purpose. Re-added.